packages/app/src/components/time-series-chart/components/patterns.tsx

```diff
@@ -0,0 +1,41 @@
+import { TimespanAnnotationConfig } from '../logic';
+
+type PatternsProps = {
+  timespanAnnotations: TimespanAnnotationConfig[] | undefined;
+  chartId: string;
+};
+
+/**
+ * For now only the estimate timespan annotation requires a pattern to be registered
+ * in the defs section, so I'm keeping that logic here.
+ *
+ * Should there be other components/annotations/etc that require these then I suggest
+ * we come up with a more generic approach to this when it becomes a requirement.
+ *
+ */
+export function Patterns({ timespanAnnotations, chartId }: PatternsProps) {
+  const estimates =
+    timespanAnnotations?.filter((x) => x.type === 'estimate') ?? [];
```

Comment:
If all you need is to know whether there are estimates, than extracting a boolean would be a bit cleaner maybe.

```suggestion
  const hasEstimateAnnotation =
    isDefined(timespanAnnotations?.find((x) => x.type === 'estimate'))
    
   return hasEsimate ? <pattern/> : null
   ```
   
   Also it returns one pattern, so I wonder why this component is called `Patterns` in plural
```

Reply:
Again, I had added a comment to the component. But I guess comments aren't read hehe. So I've renamed it.